superset/utils/core.py

```diff
@@ -189,6 +189,25 @@ class DatasourceDict(TypedDict):
     id: int
 
 
+class AdhocFilterClause(TypedDict, total=False):
```

Comment:
Since `TypedDict` doesn't support the notion of an optional property, `total=False` is required here to be able to construct partial objects based on this type. See https://github.com/python/mypy/issues/2632#issuecomment-769549870 for more info

Reply:
Hope someday this can be supported...